hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/BulkUploadHandler.java

```diff
@@ -218,18 +216,24 @@ private void readFileStream(final InputStream tempStream) {
                  * below event.
                  */
                 eventBus.publish(this, new TargetTableEvent(TargetComponentEvent.BULK_UPLOAD_PROCESS_STARTED));
+
                 while ((line = reader.readLine()) != null) {
                     innerCounter++;
                     readEachLine(line, innerCounter, totalFileSize);
                 }
-                doAssignments();
-                eventBus.publish(this, new TargetTableEvent(TargetComponentEvent.BULK_UPLOAD_COMPLETED));
 
-                // Clearing after assignments are done
-                managementUIState.getTargetTableFilters().getBulkUpload().getTargetsCreated().clear();
             } catch (final IOException e) {
                 LOG.error("Error reading file {}", tempFile.getName(), e);
+            } catch (final RuntimeException e) {
+                Optional.ofNullable(UI.getCurrent()).ifPresent(error -> UI.getCurrent().getErrorHandler()
```

Comment:
I appreciate the use of `Òptional` as a return value of a function. But I'd prefer the use of a simple if-statement here:
```
if (UI.getCurrent() != null) {
    UI.getCurrent().getErrorHandler().error(new ConnectorErrorEvent(targetTableHeader, e));
}
```